### ISSUE TITLE
Verify and document the versions of solidity supported by IoTeX mainnet

### ISSUE THREAD

**Issue by raullenchai** [iotexproject/iotex-core#2658]
IoTeX supports the Constantinople version of EVM; there are newer versions after Constantinople, which are Petersburg, Istanbul, and muirGlacier; each version of EVM supports a different range of solidity versions. For Constantinople we use, we have the impression solidity 0.4.24 - 0.5.17 roughly works but we do not know if say 0.4.20 or 0.6.8 works or not.

@Liuhaai Please do some research on this and figure out the lowest/highest versions of solidity that the current IoTeX chain supports and document the info in docs.iotex.io (@simonerom can help with that).

**Comment by raullenchai** [comment]
https://github.com/ethereum/solidity/blob/develop/liblangutil/EVMVersion.h  is helpful to understand different versions of EVM/solidity.

**Comment by Liuhaai** [comment]
## Supported solc compiler version in our chain: 0.4.20 ~ lastest(0.8.7)

**Lowest solc version: 0.4.20**

It is hard to find the documents about the lowest version of solc that supports Constantinople EVM. However, the contracts compiled from solc 0.4.20 have been verified to work on our chain.

**Highest solc version: lastest(0.8.7)**

There are some updates after constantinople EVM.

> - `constantinople`
>   
>   - Opcodes ```create2`, ``extcodehash```, `shl`, `shr` and `sar` are available in assembly.
>     
>   - Shifting operators use shifting opcodes and thus need less gas.
>     
> 
> - `petersburg`
>   
>   - The compiler behaves the same way as with constantinople.
> 
> - `istanbul`
>   
>   - Opcodes `chainid` and `selfbalance` are available in assembly.
> 
> - `berlin`
>   
>   - Gas costs for `SLOAD`, `*CALL`, `BALANCE`, `EXT*` and `SELFDESTRUCT` increased. The compiler assumes cold gas costs for such operations. This is relevant for gas estimation and the optimizer.
> 
> - `london` (**default**)
>   
>   - The block’s base fee ([EIP-3198](https://eips.ethereum.org/EIPS/eip-3198) and [EIP-1559](https://eips.ethereum.org/EIPS/eip-1559)) can be accessed via the global `block.basefee` or `basefee()` in inline assembly.

In short, our evm supports most of contracts compiled from latest version of solc, although our evm can not support several inline assembly operations(`chainid`, `selfbalance`, `basefee`) which are introduced in the later evm. This conclusion can also be verified in solidity code (https://github.com/ethereum/solidity/blob/develop/liblangutil/EVMVersion.h#L90-#L92).

In addition, solc provides `--evm-version` flag for compiling contracts in specific evm version. The code will be checked by compiler to avoid particular features or behaviours.
